subsys/bluetooth/host/hci_core.c

```diff
@@ -4670,6 +4659,17 @@ static void le_per_adv_sync_established(struct net_buf *buf)
 
 	atomic_clear_bit(pending_per_adv_sync->flags, BT_PER_ADV_SYNC_SYNCING);
 
+	err = bt_le_scan_update(false);
```

Comment:
If you get an Unexpected per adv sync established event then we won't call `bt_le_scan_update` again, is this the intention?

Reply:
Hmm, that's actually a really good point. 

If we do get an unexpected PA sync established event (which of course shouldn't happen with a proper controller ;) ), then we can either
1) Resend the PA sync create and leave scan on
2) Stop scan and call the callback and leave it up to the application
3) Ignore (current)
4) Stop scan and ignore 

The issue with 1) is that we might up in a loop with a bad controller. 

If doing 2) then we still might end up in a loop with a bad controller if the application automatically retries. Furthermore, with 2) we cannot provide a PA sync pointer if we cannot find it by handle. 

4) Could be a better approach. 

The approach of `conn` seems to be to just disconnect, without calling `bt_le_scan_update` as well, but does clear the `BT_DEV_INITIATING` flag (would require to add a similar global flag for PA if we wanted to do the same here). 

@joerchan What do you think is the proper approach here?

Reply:
> The approach of conn seems to be to just disconnect, without calling

That is a remnant of when the conn objects was assigned when we got the connected event, so if we did not have any more available and the advertiser was running, then we would just disconnect it.
In this case the controller was behaving correctly. 

In my opinion the best thing would be to stop scan and notify application of error with the callback.

Reply:
> In my opinion the best thing would be to stop scan and notify application of error with the callback.

That sounds good. In terms of providing the pointer, I almost forgot that we at any point only have 1 pending sync, so we can easily find the correct pointer :) 

I'll update the PR to reflect this.